### ISSUE TITLE
View docs source very slow on firefox

### ISSUE THREAD

**Issue by derekdreery** [rust-lang/rust#39240]
I'm not sure if this is the right place for this ticket or not, but here goes anyway.

When I view the source for any reasonably large file (maybe >600 <1000 lines), my browser freezes for about 10 seconds. I did a profile and the result is linked (spends a lot of time on 'layout', whatever that is). Just wondering if anyone else has the same issues.

`uname -r`: Linux my-hostname 4.8.13-1-ARCH #1 SMP PREEMPT Fri Dec 9 07:24:34 CET 2016 x86_64 GNU/Linux
`firefox -v`: Mozilla Firefox 50.1.0
`rustc --version`: rustc 1.14.0 (e8a012324 2016-12-16)

I'm pretty sure I've always had this issue btw, not just latest firefox/rust.

On chrome the page loads almost instantly.

![firefox_rustdoc_hang](https://cloud.githubusercontent.com/assets/1392222/22182306/274c0e52-e09a-11e6-91ab-bdbc99864180.png)

**Comment by jacwah** [comment]
No performance issues on my end. Tested with:
* MacBook Air 2014
* macOS 10.12.2
* Mozilla Firefox 46.0.1 and 50.1.0
* https://doc.rust-lang.org/std/vec/struct.Vec.html

Could you link to a page where you see this behavior?

**Comment by nagisa** [comment]
While not 10 seconds (only 4 on [Vec](https://doc.rust-lang.org/stable/src/collections/up/src/libcollections/vec.rs.html#295-298) source), I can reproduce this.

This should be reported to the Firefox folks directly, noting that removing the `font-family` from 

```
code, pre {
    font-family: "Source Code Pro", Menlo, Monaco, Consolas, "DejaVu Sans Mono", Inconsolata, monospace;
    white-space: pre-wrap;
}
```

[here](https://github.com/rust-lang/rust/blob/5a2b50b33df6015e5bae97c5224e3180885dcc66/src/doc/rust.css#L199) makes things fast again.

**Comment by jacwah** [comment]
Nevermind my previous comment, I misunderstood this issue. I get the same long freeze with:
* MacBook Air 2014
* macOS 10.12.2
* Mozilla Firefox 50.1.0
* https://doc.rust-lang.org/stable/src/collections/up/src/libcollections/vec.rs.html#295-298

**Comment by derekdreery** [comment]
I suspect it's probably better for someone from mozilla to raise it rather than me. :)

**Comment by nagisa** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1332961

**Comment by nagisa** [comment]
This is fixed in 51.0

**Comment by derekdreery** [comment]
:)
